Feature 503 fix input dir

hankenstein2 opened the pull request:
Added requirements and logged errors for missing *_INPUT_TEMPLATE and changed *_INPUT_DIR to be empty string if not found.

Commit:
Added error logging if *INPUT_TEMPLATE does not exist
and changed *_INPUT_DIR to default to ''

Commit:
Fixed formatting issue

Commit:
Added check for empty TC_STAT_INPUT_DIR

georgemccabe commented on metplus/wrappers/ascii2nc_wrapper.py:
```diff
@@ -48,6 +48,8 @@ def create_c_dict(self):
         c_dict['OBS_INPUT_TEMPLATE'] = self.config.getraw('filename_templates',
                                                           'ASCII2NC_INPUT_TEMPLATE')
+        if not c_dict['OBS_INPUT_TEMPLATE']:
+            self.log_error("OBS_INPUT_TEMPLATE required to run")
```
You should use the configuration variable name, not the c_dict name.
```suggestion
            self.log_error("ASCII2NC_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/compare_gridded_wrapper.py:
```diff
@@ -128,6 +128,8 @@ def read_climo_wrapper_specific(self, met_tool, c_dict):
                                    f'{met_tool}_CLIMO_{climo_item}_INPUT_TEMPLATE',
                                    '')
+            if not c_dict[f'CLIMO_{climo_item}_INPUT_TEMPLATE']:
+                self.log_error("f'CLIMO_{climo_item}_INPUT_TEMPLATE' needed to run")
```
```suggestion
                self.log_error("f'{met_tool}_CLIMO_{climo_item}_INPUT_TEMPLATE' needed to run")
```

georgemccabe commented on metplus/wrappers/ensemble_stat_wrapper.py:
```diff
@@ -106,13 +106,20 @@ def create_c_dict(self):
                              'OBS_ENSEMBLE_STAT_POINT_INPUT_TEMPLATE')
 
+        if not c_dict['OBS_POINT_INPUT_TEMPLATE']:
+            self.log_error("OBS_POINT_INPUT_TEMPLATE requrired to run")
```
```suggestion
            self.log_error("OBS_ENSEMBLE_STAT_POINT_INPUT_TEMPLATE requrired to run")
```

georgemccabe commented on metplus/wrappers/ensemble_stat_wrapper.py:
```diff
@@ -106,13 +106,20 @@ def create_c_dict(self):
                              'OBS_ENSEMBLE_STAT_GRID_INPUT_TEMPLATE')
 
+        if not c_dict['OBS_GRID_INPUT_TEMPLATE']:
+            self.log_error("OBS_GRID_INPUT_TEMPLATE requrired to run")
```
```suggestion
            self.log_error("OBS_ENSEMBLE_STAT_GRID_INPUT_TEMPLATE requrired to run")
```

georgemccabe committed:
Example wrapper warning if no template

Changed to warning instead of error because it can still demonstrate how time looping works.

georgemccabe commented on metplus/wrappers/grid_stat_wrapper.py:
```diff
@@ -39,18 +39,25 @@ def create_c_dict(self):
           self.config.getraw('filename_templates',
                              'OBS_GRID_STAT_INPUT_TEMPLATE')
+        if not c_dict['OBS_INPUT_TEMPLATE']:
+            self.log_error("OBS_INPUT_TEMPLATE required to run")
```
```suggestion
            self.log_error("OBS_GRID_STAT_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/grid_stat_wrapper.py:
```diff
@@ -39,18 +39,25 @@ def create_c_dict(self):
                              'FCST_GRID_STAT_INPUT_TEMPLATE')
+
+        if not c_dict['FCST_INPUT_TEMPLATE']:
+            self.log_error("FCST_INPUT_TEMPLATE required to run")
```
```suggestion
            self.log_error("FCST_GRID_STAT_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/mtd_wrapper.py:
```diff
@@ -55,6 +55,9 @@ def create_c_dict(self):
               self.config.getraw('filename_templates',
                                  'FCST_MTD_INPUT_TEMPLATE')
+            if not c_dict['FCST_INPUT_TEMPLATE']:
+                self.log_error("FCST_INPUT_TEMPLATE required to run")
```
```suggestion
                self.log_error("FCST_MTD_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/mtd_wrapper.py:
```diff
@@ -87,6 +90,8 @@ def create_c_dict(self):
               self.config.getraw('filename_templates',
                                    'OBS_MTD_INPUT_TEMPLATE')
+            if not c_dict['OBS_INPUT_TEMPLATE']:
+                self.log_error("OBS_INPUT_TEMPLATE required to run")
```
```suggestion
                self.log_error("OBS_MTD_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/pcp_combine_wrapper.py:
```diff
@@ -95,6 +95,8 @@ def set_fcst_or_obs_dict_items(self, d_type, c_dict):
         c_dict[d_type+'_INPUT_TEMPLATE'] = self.config.getraw('filename_templates',
                                                               d_type+'_PCP_COMBINE_INPUT_TEMPLATE', '')
+        if not c_dict[d_type+'_INPUT_TEMPLATE']:
+            self.log_error(d_type+"_INPUT_TEMPLATE required to run")
```
```suggestion
            self.log_error(d_type + "_PCP_COMBINE_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/point2grid_wrapper.py:
```diff
@@ -49,6 +49,9 @@ def create_c_dict(self):
                                                           'POINT_2_GRID_INPUT_TEMPLATE')
 
+        if not c_dict['OBS_INPUT_TEMPLATE']:
+            self.log_error("OBS_INPUTE_TEMPLATE required to run")
```
```suggestion
            self.log_error("POINT_2_GRID_INPUT_TEMPLATE required to run")
```

georgemccabe committed:
template was set above new addition

moved check up, removed extra getraw call, and changed log to use config var names

georgemccabe commented on metplus/wrappers/series_analysis_wrapper.py:
```diff
@@ -76,6 +76,9 @@ def create_c_dict(self):
                                  f'{data_type}_SERIES_ANALYSIS_INPUT_TEMPLATE',
                                  '')
+            if not c_dict[f'{data_type}_INPUT_TEMPLATE']:
+                self.log_error(f"{data_type}_INPUT_TEMPLATE required to run")
```
```suggestion
                self.log_error(f"{data_type}_SERIES_ANALYSIS_INPUT_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/tc_pairs_wrapper.py:
```diff
@@ -125,11 +125,17 @@ def create_c_dict(self):
                                                       'TC_PAIRS_ADECK_TEMPLATE',
                                                       '')
+        if not c_dict['ADECK_TEMPLATE']:
+            self.log_error("ADECK_TEMPLATE required to run")
```
```suggestion
            self.log_error("TC_PAIRS_ADECK_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/tc_pairs_wrapper.py:
```diff
@@ -125,11 +125,17 @@ def create_c_dict(self):
         c_dict['BDECK_TEMPLATE'] = self.config.getraw('filename_templates',
                                                       'TC_PAIRS_BDECK_TEMPLATE')
+        if not c_dict['BDECK_TEMPLATE']:
+            self.log_error("BDECK_TEMPLATE required to run")
```
```suggestion
            self.log_error("TC_PAIRS_BDECK_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/tc_pairs_wrapper.py:
```diff
@@ -125,11 +125,17 @@ def create_c_dict(self):
                                                       'TC_PAIRS_EDECK_TEMPLATE',
                                                       '')
+        if not c_dict['EDECK_TEMPLATE']:
+            self.log_error("EDECK_TEMPLATE required to run")
```
```suggestion
            self.log_error("TC_PAIRS_EDECK_TEMPLATE required to run")
```

georgemccabe commented on metplus/wrappers/tc_stat_wrapper.py:
```diff
@@ -229,7 +229,10 @@ def create_c_dict(self):
-        c_dict['INPUT_DIR'] = self.config.getdir('TC_STAT_INPUT_DIR')
+        c_dict['INPUT_DIR'] = self.config.getdir('TC_STAT_INPUT_DIR','')
+
+        if c_dict['INPUT_DIR'] == '':
```
```suggestion
        if not c_dict['INPUT_DIR']:
```

georgemccabe reviewed (changes requested):
Errors should reference the configuration variable name instead of the c_dict value so it is clear what the user has to set. I made the changes so you should be able to accept the suggestions (unless I made a typo!).

hankenstein2 committed:
Update metplus/wrappers/ascii2nc_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/ensemble_stat_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/grid_stat_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/series_analysis_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/compare_gridded_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/ensemble_stat_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/tc_pairs_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/tc_pairs_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/tc_pairs_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/tc_stat_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/grid_stat_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/mtd_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/mtd_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/pcp_combine_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 committed:
Update metplus/wrappers/point2grid_wrapper.py

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

hankenstein2 commented:
Did not see any typos and I accepted/committed all the changes.

Hank

On Mon, Jun 15, 2020 at 2:11 PM George McCabe <notifications@github.com>
wrote:

> *@georgemccabe* requested changes on this pull request.
>
> Errors should reference the configuration variable name instead of the
> c_dict value so it is clear what the user has to set. I made the changes so
> you should be able to accept the suggestions (unless I made a typo!).
> ------------------------------
>
> In metplus/wrappers/ascii2nc_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440411881>:
>
> > @@ -48,6 +48,8 @@ def create_c_dict(self):
>
>          c_dict['OBS_INPUT_DIR'] = self.config.getdir('ASCII2NC_INPUT_DIR', '')
>
>          c_dict['OBS_INPUT_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                            'ASCII2NC_INPUT_TEMPLATE')
>
> +        if not c_dict['OBS_INPUT_TEMPLATE']:
>
> +            self.log_error("OBS_INPUT_TEMPLATE required to run")
>
>
> You should use the configuration variable name, not the c_dict name.
> ⬇️ Suggested change
>
> -            self.log_error("OBS_INPUT_TEMPLATE required to run")
>
> +            self.log_error("ASCII2NC_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/compare_gridded_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440412110>:
>
> > @@ -128,6 +128,8 @@ def read_climo_wrapper_specific(self, met_tool, c_dict):
>
>                  self.config.getraw('filename_templates',
>
>                                     f'{met_tool}_CLIMO_{climo_item}_INPUT_TEMPLATE',
>
>                                     '')
>
> +            if not c_dict[f'CLIMO_{climo_item}_INPUT_TEMPLATE']:
>
> +                self.log_error("f'CLIMO_{climo_item}_INPUT_TEMPLATE' needed to run")
>
>
> ⬇️ Suggested change
>
> -                self.log_error("f'CLIMO_{climo_item}_INPUT_TEMPLATE' needed to run")
>
> +                self.log_error("f'{met_tool}_CLIMO_{climo_item}_INPUT_TEMPLATE' needed to run")
>
>
> ------------------------------
>
> In metplus/wrappers/ensemble_stat_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440412257>:
>
> > @@ -106,13 +106,20 @@ def create_c_dict(self):
>
>            self.config.getraw('filename_templates',
>
>                               'OBS_ENSEMBLE_STAT_POINT_INPUT_TEMPLATE')
>
>
>
> +        if not c_dict['OBS_POINT_INPUT_TEMPLATE']:
>
> +            self.log_error("OBS_POINT_INPUT_TEMPLATE requrired to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("OBS_POINT_INPUT_TEMPLATE requrired to run")
>
> +            self.log_error("OBS_ENSEMBLE_STAT_POINT_INPUT_TEMPLATE requrired to run")
>
>
> ------------------------------
>
> In metplus/wrappers/ensemble_stat_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440412352>:
>
> >          c_dict['OBS_GRID_INPUT_DIR'] = \
>
>            self.config.getdir('OBS_ENSEMBLE_STAT_GRID_INPUT_DIR', '')
>
>
>
>          c_dict['OBS_GRID_INPUT_TEMPLATE'] = \
>
>            self.config.getraw('filename_templates',
>
>                               'OBS_ENSEMBLE_STAT_GRID_INPUT_TEMPLATE')
>
>
>
> +        if not c_dict['OBS_GRID_INPUT_TEMPLATE']:
>
> +            self.log_error("OBS_GRID_INPUT_TEMPLATE requrired to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("OBS_GRID_INPUT_TEMPLATE requrired to run")
>
> +            self.log_error("OBS_ENSEMBLE_STAT_GRID_INPUT_TEMPLATE requrired to run")
>
>
> ------------------------------
>
> In metplus/wrappers/grid_stat_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440414386>:
>
> >          c_dict['OBS_INPUT_TEMPLATE'] = \
>
>            self.config.getraw('filename_templates',
>
>                               'OBS_GRID_STAT_INPUT_TEMPLATE')
>
> +        if not c_dict['OBS_INPUT_TEMPLATE']:
>
> +            self.log_error("OBS_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("OBS_INPUT_TEMPLATE required to run")
>
> +            self.log_error("OBS_GRID_STAT_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/grid_stat_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440414497>:
>
> >          c_dict['FCST_INPUT_TEMPLATE'] = \
>
>            self.config.getraw('filename_templates',
>
>                               'FCST_GRID_STAT_INPUT_TEMPLATE')
>
> +
>
> +        if not c_dict['FCST_INPUT_TEMPLATE']:
>
> +            self.log_error("FCST_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("FCST_INPUT_TEMPLATE required to run")
>
> +            self.log_error("FCST_GRID_STAT_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/mtd_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440414593>:
>
> > @@ -55,6 +55,9 @@ def create_c_dict(self):
>
>              c_dict['FCST_INPUT_TEMPLATE'] = \
>
>                self.config.getraw('filename_templates',
>
>                                   'FCST_MTD_INPUT_TEMPLATE')
>
> +            if not c_dict['FCST_INPUT_TEMPLATE']:
>
> +                self.log_error("FCST_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -                self.log_error("FCST_INPUT_TEMPLATE required to run")
>
> +                self.log_error("FCST_MTD_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/mtd_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440414671>:
>
> > @@ -87,6 +90,8 @@ def create_c_dict(self):
>
>              c_dict['OBS_INPUT_TEMPLATE'] = \
>
>                self.config.getraw('filename_templates',
>
>                                     'OBS_MTD_INPUT_TEMPLATE')
>
> +            if not c_dict['OBS_INPUT_TEMPLATE']:
>
> +                self.log_error("OBS_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -                self.log_error("OBS_INPUT_TEMPLATE required to run")
>
> +                self.log_error("OBS_MTD_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/pcp_combine_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440414967>:
>
> > @@ -95,6 +95,8 @@ def set_fcst_or_obs_dict_items(self, d_type, c_dict):
>
>          c_dict[d_type+'_INPUT_DIR'] = self.config.getdir(d_type+'_PCP_COMBINE_INPUT_DIR', '')
>
>          c_dict[d_type+'_INPUT_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                                d_type+'_PCP_COMBINE_INPUT_TEMPLATE', '')
>
> +        if not c_dict[d_type+'_INPUT_TEMPLATE']:
>
> +            self.log_error(d_type+"_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error(d_type+"_INPUT_TEMPLATE required to run")
>
> +            self.log_error(d_type + "_PCP_COMBINE_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/point2grid_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440415085>:
>
> > @@ -49,6 +49,9 @@ def create_c_dict(self):
>
>          c_dict['OBS_INPUT_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                            'POINT_2_GRID_INPUT_TEMPLATE')
>
>
>
> +        if not c_dict['OBS_INPUT_TEMPLATE']:
>
> +            self.log_error("OBS_INPUTE_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("OBS_INPUTE_TEMPLATE required to run")
>
> +            self.log_error("POINT_2_GRID_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/series_analysis_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440418516>:
>
> > @@ -76,6 +76,9 @@ def create_c_dict(self):
>
>                self.config.getraw('filename_templates',
>
>                                   f'{data_type}_SERIES_ANALYSIS_INPUT_TEMPLATE',
>
>                                   '')
>
> +            if not c_dict[f'{data_type}_INPUT_TEMPLATE']:
>
> +                self.log_error(f"{data_type}_INPUT_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -                self.log_error(f"{data_type}_INPUT_TEMPLATE required to run")
>
> +                self.log_error(f"{data_type}_SERIES_ANALYSIS_INPUT_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/tc_pairs_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440418723>:
>
> > @@ -125,11 +125,17 @@ def create_c_dict(self):
>
>          c_dict['ADECK_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                        'TC_PAIRS_ADECK_TEMPLATE',
>
>                                                        '')
>
> +        if not c_dict['ADECK_TEMPLATE']:
>
> +            self.log_error("ADECK_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("ADECK_TEMPLATE required to run")
>
> +            self.log_error("TC_PAIRS_ADECK_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/tc_pairs_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440418835>:
>
> >          c_dict['BDECK_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                        'TC_PAIRS_BDECK_TEMPLATE')
>
> +        if not c_dict['BDECK_TEMPLATE']:
>
> +            self.log_error("BDECK_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("BDECK_TEMPLATE required to run")
>
> +            self.log_error("TC_PAIRS_BDECK_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/tc_pairs_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440418934>:
>
> >          c_dict['EDECK_TEMPLATE'] = self.config.getraw('filename_templates',
>
>                                                        'TC_PAIRS_EDECK_TEMPLATE',
>
>                                                        '')
>
> +        if not c_dict['EDECK_TEMPLATE']:
>
> +            self.log_error("EDECK_TEMPLATE required to run")
>
>
> ⬇️ Suggested change
>
> -            self.log_error("EDECK_TEMPLATE required to run")
>
> +            self.log_error("TC_PAIRS_EDECK_TEMPLATE required to run")
>
>
> ------------------------------
>
> In metplus/wrappers/tc_stat_wrapper.py
> <https://github.com/NCAR/METplus/pull/507#discussion_r440419084>:
>
> > @@ -229,7 +229,10 @@ def create_c_dict(self):
>
>
>
>          c_dict['MET_INSTALL_DIR'] = self.config.getdir('MET_INSTALL_DIR')
>
>
>
> -        c_dict['INPUT_DIR'] = self.config.getdir('TC_STAT_INPUT_DIR')
>
> +        c_dict['INPUT_DIR'] = self.config.getdir('TC_STAT_INPUT_DIR','')
>
> +
>
> +        if c_dict['INPUT_DIR'] == '':
>
>
> ⬇️ Suggested change
>
> -        if c_dict['INPUT_DIR'] == '':
>
> +        if not c_dict['INPUT_DIR']:
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/METplus/pull/507#pullrequestreview-430948639>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHAUIKAG3TYRDHMO7NVAKNTRWZ6A5ANCNFSM4N6Q6LLA>
> .
>


-- 
--------------------------------------------------------------
Hank Fisher       Research Applications Laboratory
303-497-2817      National Center for Atmospheric Research
fisherh@ucar.edu  P.O. Box 3000, Boulder CO 80307-3000
--------------------------------------------------------------

georgemccabe commented:
Ah, it looks like some of the use cases failed on Travis. I looks like some of the templates are not required to run the use case, so we need to remove the error check.

The CLIMO_ variables are an example. I will do another review to find the ones that are optional.

georgemccabe committed:
climo templates are not required to be set

georgemccabe committed:
remove check for optional obs grid/point templates

georgemccabe committed:
MTD can be run on just fcst or obs, so

template is not required

georgemccabe committed:
not all deck templates required to run

georgemccabe reviewed (approved):
I committed some changes because some of the templates are not actually required to run. They are optional. These commits should trigger the Travis build, so if the tests all run successfully, then we can merge these changes. Sorry to mislead!

hankenstein2 commented:
No worries, always good for me to go through a different part of the code.
I wondered about the *DECK ones but did not go through the examples. I
think I rely too much on the unit tests.

On Mon, Jun 15, 2020 at 3:28 PM George McCabe <notifications@github.com>
wrote:

> *@georgemccabe* approved this pull request.
>
> I committed some changes because some of the templates are not actually
> required to run. They are optional. These commits should trigger the Travis
> build, so if the tests all run successfully, then we can merge these
> changes. Sorry to mislead!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/METplus/pull/507#pullrequestreview-431006335>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHAUIKCU4BEQJFUBBQIDBDDRW2G6ZANCNFSM4N6Q6LLA>
> .
>


-- 
--------------------------------------------------------------
Hank Fisher       Research Applications Laboratory
303-497-2817      National Center for Atmospheric Research
fisherh@ucar.edu  P.O. Box 3000, Boulder CO 80307-3000
--------------------------------------------------------------

georgemccabe committed:
fixed bug in SeriesAnalysis checking templates

georgemccabe commented:
All tests passed now, so merging changes into develop.
